File: src/client/xnative/lion.xdroid/Service/RefreshTokenService.cs
using Android.App;
using Android.Util;
using Lion.XDroid;
using Firebase.Iid;
using Lion.XDroid.Libs;
using System.Collections.Generic;
using System;

namespace Lion.XDroid.Service
{
    [Service]
    [IntentFilter(new[] { "com.google.firebase.INSTANCE_ID_EVENT" })]
    public class RefreshTokenService : FirebaseInstanceIdService
    {
        public override void OnTokenRefresh()
        {
            var _app_cache = new AppPreferences(this.ApplicationContext);
            var _user_token = _app_cache.UserTokenKey;

            if (String.IsNullOrEmpty(_user_token) == false)
            {
                var _refreshed_id = FirebaseInstanceId.Instance.Token;
                this.UpdateDeviceId(_user_token, _refreshed_id);
            }
        }

        private void UpdateDeviceId(string user_token, string device_id)
        {
            try
            {
                var parameter = new Dictionary<string, string>();
                {
                    parameter.Add("device_type", "A");
                    parameter.Add("device_id", device_id);
                }

                var api = new ApiAsnycTask(this, GetString(Resource.String.api_url) + "user/UpdateDeviceId", user_token);
                api.Execute(parameter);

                api.SendFinish += (s, e) =>
                {
                };
            }
            catch (Java.Lang.Exception ex)
            {
                Log.Error(this.GetType().Name, ex.Message);
            }
        }
    }
}

File: src/client/xcode/Lion.Droid/app/src/main/java/com/odinsoftware/lottolion/service/RefreshTokenService.java
package com.odinsoftware.lottolion.service;

import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.iid.FirebaseInstanceIdService;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;
import com.odinsoftware.lottolion.R;
import com.odinsoftware.lottolion.type.ApiResult;

import java.util.HashMap;

/**
 * Created by limyg on 2017-04-27.
 */
public class RefreshTokenService extends FirebaseInstanceIdService {
    @Override
    public void onTokenRefresh() {
        String refreshedId = FirebaseInstanceId.getInstance().getToken();
        String token = LLSharedPreferences.LL(getApplicationContext()).getSaveSession();
        if(LLCommon.Get().CheckExpired(token) == false){
            UpdateDeviceId(token, refreshedId);
        }
    }
    private void UpdateDeviceId(String key, String id){
        try{
            HashMap<String, String> parameter = new HashMap<>();
            parameter.put("device_type","A");
            parameter.put("device_id",id);
            ApiAsnycTask api = new ApiAsnycTask(getString(R.string.api_url)+"user/UpdateDeviceId",key);
            api.execute(parameter);
        }
        catch (Exception ex){

        }
    }
}
